text/dependencies/rfcs/0000-dependency-management-phase-one.md

```diff
@@ -0,0 +1,287 @@
+| rust              | https://static.rust-lang.org/dist/                        | yes, buildpacks-ci                       |
+| tini              | https://github.com/krallin/tini/tarball                   | yes, buildpacks-ci                       |
+| yarn              | https://github.com/yarnpkg/yarn/releases/download/v1.15.2 | processed, buildpacks-ci                 |
+</details>
+
+### Transfer Compilation Code
+
+For any dependencies that must still be compiled or processed in some way, the
+code from the Cloud Foundry repositories should be rewritten,
+and moved over to the buildpack under a directory called
+`dependency/compilation`. The code should be easily runnable from a local
+machine to compile the code, and should be documented with a README about how
+to use it. The dependency will likely need to be compiled against different
+stacks/architectures, so this should also be enumerated in the RFC and taken
+account of in the new compilation code.
+
+Maintainers should also consider if the dependency of interest if OS
+distribution-agnostic, or if it will need to be compiled separately depending
+on the distribution or platform it's used on.
+
+The code will eventually be used in a Github Actions workflow, so the location
+needs to be standardized across buildpacks. The workflows and Github Actions to
+use this code will be enumerated in a separate RFC.
+
+### Bucket Setup
+
+The new dependency management automation (described in a subsequent RFC) will
+rely on the `buildpack.toml` and dependency-specific code as the source of
+truth for the latest metadata and known versions. This will lead to greater
+transparency and discoverability in how we update dependencies.
+
+Since metadata and known versions won't be stored data anymore, buckets will
+only be needed to store dependencies that have been compiled or processed in
+some way. For these cases, a dependency-specific bucket set up in Paketo
+Dependencies project within GCP to store data. The directory name will be the
+name of the dependency and it will contain dependency archives (if compiled).
+Push credentials to this bucket will be reserved for the buildpack and
+maintainers, but the contents will be publicly available.
+
+### Remove Known Versions
+
+As mentioned above, dependency workflows will use the `buildpack.toml` as a
+source of the latest versions we support. Because of this change, we will no
+longer need to keep track of all known versions in a separate file.
+
+### Transfer Version Retrieval Code
+
+Version discovery code comes from
+[dep-server/pkg/dependency](https://github.com/paketo-buildpacks/dep-server/tree/main/pkg/dependency).
+Each dependency has a slightly different way new versions are made
+discoverable. This code should be ported from the dep-server repository to the
+buildpack under a directory named `dependency/retrieval`. It should be
+well-documented and useable on a local environment. The location must be
+standardized for use in automation.
```

Comment:
Are we planning on making a library for common abstracted features?

I'm ok with deferring this decision until we start rolling out the changes as it will be easier to reason about the boundaries of common abstractions once we start to separate the logic from the single existing location.

Reply:
I see the comment below about a dep-server library for licenses/pURL info. If we're planning on starting there, that would be a good place to add any abstracted retrieval behavior.

Reply:
I think eventually we can, and when we actually get to implementing it may become obvious that we need some common library functions. For the first go-around it's probably find to leave off.

Reply:
I'll add a note about adding other potential abstracted logic to the same dep-server library though.

Reply:
@joshuatcasey made the same comment to me, that we should make an effort to do this in the future so our implementations of other parts of the process are uniform/we can save time in each buildpack. Agreed!

Reply:
Sure thing. It might actually be confusing to have all of this stuff live in the dep-server.